ddtrace/internal/telemetry/writer.py

```diff
@@ -145,7 +147,7 @@ def add_integration(self, integration_name, auto_enabled):
         :param bool auto_enabled: True if module is enabled in _monkey.PATCH_MODULES
         """
         with self._lock:
-            if not self._enabled:
+            if self._enabled is not None and not self._enabled:
```

Comment:
should we also be checking for `self._enabled` in `periodic`? I know this is an aside to this PR

Reply:
I believe Periodic is only called when the periodic service is started. The current flow is: 

TelemetryWriter.enable() ->  TelemetryWriter.start() -> TelemetryWriter.periodic() (called every 60 seconds)